File: 35-SearchInsertPosition/35-SearchInsertPosition.js
// Last updated: 24/07/2026, 23:54:55
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var searchInsert = function (arr, target) {
    let l = 0, r = arr.length - 1;
    while (l <= r) {
        let m = (l + r) >> 1;
        if (arr[m] === target) return m;
        if (arr[m] > target) r = m - 1; else l = m + 1;
    }
    return l;
};

File: 35-SearchInsertPosition/35-SearchInsertPosition.java
// Last updated: 24/07/2026, 23:55:03
class Solution {
    public int searchInsert(int[] nums, int target) {
        int first=0;
        int last=nums.length-1;
  
        while(first<=last){
            int b=first+(last-first)/2;
            if(nums[b]==target){
                return b;
            }else if(nums[b]>target){
                last = b-1;
            }else {
                first=b+1;
                
            }
        }
      return last+1;
        
    }
        
}